gensim/downloader.py

```diff
@@ -0,0 +1,437 @@
+    data_folder_dir = os.path.join(base_dir, name)
+    tmp_dir = tempfile.mkdtemp()
+    init_path = os.path.join(tmp_dir, "__init__.py")
+    urllib.urlretrieve(url_load_file, init_path)
+    total_parts = _get_parts(name)
+    if total_parts > 1:
+        concatenated_folder_name = "{f}.gz".format(f=name)
+        concatenated_folder_dir = os.path.join(tmp_dir, concatenated_folder_name)
+        for part in range(0, total_parts):
+            url_data = \
+                "https://github.com/RaRe-Technologies/gensim-data/releases/download/{f}/{f}.gz_0{p}" \
+                .format(f=name, p=part)
+
+            fname = "{f}.gz_0{p}".format(f=name, p=part)
+            dst_path = os.path.join(tmp_dir, fname)
+            urllib.urlretrieve(
+                url_data, dst_path,
+                reporthook=partial(_progress, part=part, total_parts=total_parts)
+            )
+            if _calculate_md5_checksum(dst_path) == _get_checksum(name, part):
+                sys.stdout.write("\n")
+                sys.stdout.flush()
+                logger.info("Part %s/%s downloaded", part + 1, total_parts)
+            else:
+                shutil.rmtree(tmp_dir)
+                raise Exception("Checksum comparison failed, try again")
+        with open(concatenated_folder_dir, 'wb') as wfp:
+            for part in range(0, total_parts):
+                part_path = os.path.join(tmp_dir, "{f}.gz_0{p}".format(f=name, p=part))
+                with open(part_path, "rb") as rfp:
+                    shutil.copyfileobj(rfp, wfp)
+                os.remove(part_path)
+        os.rename(tmp_dir, data_folder_dir)
+    else:
+        url_data = "https://github.com/RaRe-Technologies/gensim-data/releases/download/{f}/{f}.gz".format(f=name)
+        fname = "{f}.gz".format(f=name)
+        dst_path = os.path.join(tmp_dir, fname)
+        urllib.urlretrieve(url_data, dst_path, reporthook=_progress)
+        if _calculate_md5_checksum(dst_path) == _get_checksum(name):
+            sys.stdout.write("\n")
+            sys.stdout.flush()
+            logger.info("%s downloaded", name)
+        else:
+            shutil.rmtree(tmp_dir)
+            raise Exception("Checksum comparison failed, try again")
+        os.rename(tmp_dir, data_folder_dir)
+
+
+def _get_filename(name):
+    """Retrieve the filename of the dataset/model.
+
+    Parameters
+    ----------
+    name: str
+        Name of dataset/model.
+
+    Returns
+    -------
+    str:
+        Filename of the dataset/model.
+
+    """
+    information = info()
+    corpora = information['corpora']
+    models = information['models']
+    if name in corpora:
+        return information['corpora'][name]["file_name"]
+    elif name in models:
+        return information['models'][name]["file_name"]
+
+
+def load(name, return_path=False):
+    """Download (if needed) dataset/model and load it to memory (managed by return_path).
```

Comment:
"managed by return_path" doesn't sound correct. Do you mean "unless return_path is set"?

Reply:
Yes, that's true.